### PR TITLE
Queries converted to complete relay connection model

### PR DESCRIPTION
**Pull request by fgm**

### CODE REVIEW

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php [hunk 1]
```diff
@@ -144,16 +170,20 @@ public function resolve($parent, array $args = NULL, ResolveInfo $info) {
...
+    }
+    else {
       $entities = $storage->loadMultiple($result);
 
-      // Filter entities that the current user doesn't have view access for.
-      return array_filter($entities, function (EntityInterface $entity) {
-        return $entity->access('view');
-      });
+      $filteredEntities = array_filter($entities,
+        function (EntityInterface $entity) {
```
Comment (fubhy) [inline src/GraphQL/Field/Root/Entity/EntityQueryField.php:182]: We should check for AccessibleInterface here just for good measure.

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php [hunk 2]
```diff
@@ -144,16 +170,20 @@ public function resolve($parent, array $args = NULL, ResolveInfo $info) {
...
+          return $entity->access('view');
+        }
+      );
     }
 
-    return [];
+    return ArrayConnection::connectionFromArray($filteredEntities, $args);
```
Comment (fubhy) [inline src/GraphQL/Field/Root/Entity/EntityQueryField.php:188]: This is fine as an interim solution but we need to do the Connection logic on the query level to make this feasible for production.

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php [hunk 3]
```diff
@@ -7,14 +7,20 @@
...
 use Drupal\Core\Entity\TypedData\EntityDataDefinitionInterface;
 use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\Core\TypedData\TypedDataManagerInterface;
 use Drupal\graphql\GraphQL\Field\FieldBase;
+use Drupal\graphql\GraphQL\Type\Entity\EntityObjectType;
+use Drupal\graphql\GraphQL\Type\EntityType\EntityTypeObjectType;
 use Drupal\graphql\TypeResolver\TypeResolverInterface;
 use Drupal\graphql\Utility\StringHelper;
+use Examples\StarWars\TestDataProvider;
```
Comment (fubhy) [inline src/GraphQL/Field/Root/Entity/EntityQueryField.php]: This is a leftover from testing I guess?

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php [hunk 4]
```diff
@@ -7,14 +7,20 @@
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Entity\TypedData\EntityDataDefinitionInterface;
 use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\Core\TypedData\TypedDataManagerInterface;
 use Drupal\graphql\GraphQL\Field\FieldBase;
+use Drupal\graphql\GraphQL\Type\Entity\EntityObjectType;
+use Drupal\graphql\GraphQL\Type\EntityType\EntityTypeObjectType;
```
Comment (fubhy) [inline src/GraphQL/Field/Root/Entity/EntityQueryField.php]: Unused use statement.

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php (hunk 2 above)
Comment (fgm) [reply src/GraphQL/Field/Root/Entity/EntityQueryField.php:188]: Probably sufficient for config entities but not content entities, indeed

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php (hunk 3 above)
Comment (fgm) [reply src/GraphQL/Field/Root/Entity/EntityQueryField.php]: oops

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php (hunk 4 above)
Comment (fgm) [reply src/GraphQL/Field/Root/Entity/EntityQueryField.php]: removed

File: src/GraphQL/Field/Root/Entity/EntityQueryField.php (hunk 1 above)
Comment (fgm) [reply src/GraphQL/Field/Root/Entity/EntityQueryField.php:182]: I don’t think the AccessibleInterface check is needed: it’s already type-hinted on EntityInterface, which extends AccessibleInterface

**Comment by fubhy** [comment]
I removed the last commit and merged this manually.

**Pull request closed without merging**